### PR TITLE
[OSDEV-888] Fix an error that occurs when trying to open a facility from the Status Reports page

### PR DESCRIPTION
**Pull request by mazursasha1990**
[OSDEV-888](https://opensupplyhub.atlassian.net/browse/OSDEV-888): Fix an error that occurs when trying to open a facility from the Status Reports page

The error occurred due to activity reports with the status `pending` containing fields with `null` values and these values pass to the `format_date` function as an argument.
Modified the `get_activity_reports` method in the `FacilityIndexDetailsSerializer` to prevent passing a falsy `date` argument into the `format_date` function.

### CODE REVIEW

File: src/django/api/serializers/facility/utils.py [hunk 1]
```diff
@@ -410,6 +410,9 @@ def regroup_claims_for_sector_field(claims: list,
 
 
 def format_date(date: str) -> str:
+    if not date:
```
Comment (killev) [inline src/django/api/serializers/facility/utils.py:413]: I doubt that None passed here is a normal situation. I think we need to dig deeper why it happened here. I expect a problem with logic in some other place

File: src/django/api/serializers/facility/facility_index_details_serializer.py [hunk 2]
```diff
@@ -190,26 +190,29 @@ def get_claim_info(self, facility):
     def get_activity_reports(self, facility):
         return [
             {
-                'facility': item['facility'] or None,
-                'reported_by_contributor': (
-                    item['reported_by_contributor'] or None
+                'facility': item.get('facility'),
```
Comment (killev) [inline src/django/api/serializers/facility/facility_index_details_serializer.py:193]: What's the difference between item['facility'] and item.get('facility')?

File: src/django/api/tests/test_facility_index_details_serializer.py [hunk 3]
```diff
@@ -348,3 +353,30 @@ def test_get_other_names_embed_mode_active(self):
         expected_other_names = []
         actual_other_names = serializer.get_other_names(facility_index)
         self.assertEqual(expected_other_names, actual_other_names)
+
+    def test_get_activity_reports(self):
+        FacilityActivityReport.objects.create(
```
Comment (roman-stolar) [inline src/django/api/tests/test_facility_index_details_serializer.py:358]: Did test covered a scenario that was reproduced on this bug?

Comment (mazursasha1990) [reply src/django/api/tests/test_facility_index_details_serializer.py:358]: Yes, it does.

**Review by killev — DISMISSED** [review]
LGTM, Approved

**Review by roman-stolar — APPROVED** [review]
LGTM

**Review by killev — APPROVED** [review]
LGTM, Approved

**Pull request merged**